mypy/checkmember.py

```diff
@@ -1420,5 +1458,45 @@ def analyze_decorator_or_funcbase_access(
     if isinstance(defn, Decorator):
         return analyze_var(name, defn.var, itype, mx)
     typ = function_type(defn, mx.chk.named_type("builtins.function"))
+    is_trivial_self = False
```

Comment:
Would it make sense to introduce an interface implemented by `Decorator` and `FuncBase` and put an `is_trivial_self()` property/method there? I remember seeing similar decorator/func ladders for other reasons too, so such refactoring won't be wasted - they should have some common properties, after all. (and this block is already copy-pasted twice)

Reply:
Yeah, I was not sure about this, ultimately I decided to save some microseconds by "inlining" it.

Reply:
Would you mind me profiling this difference (inline vs shared interface) tomorrow? I wanted to figure out how to profile mypyc-compiled code reliably, sounds like a good chance. (Anyway this isn't blocking, this part can be refactored separately later - perhaps together with some other "shared" methods)

Reply:
I definitely can't stop you, but I guess this may be tricky, as difference is likely small.

Reply:
Likely non-existent when running, say, whole self-check, but easy to microbenchmark! (but probably it'd be better to profile some reasonable refactoring attempt instead)

Reply:
FWIW, I've been working on a script to simplify profiling mypyc-compiled mypy. The current version I have uses py-spy and doesn't work on 3.12 and later, though. I'll probably switch to `perf` instead (Linux only).

Reply:
Yep, I'm actually looking at `perf` for this purpose since I'm already familiar with it:)

Reply:
Anyway, I don't want to wait long time for more precise measurements here (since the perf win is obvious). Unless there are objections, I will merge this soon (you can still profile and fine tune this later).